src/cmd/logic-function.js

```diff
@@ -46,9 +52,129 @@ module.exports = class LogicFunctionsCommand extends CLICommandBase {
 		console.log(org, name, id);
 	}
 
-	async create({ org, params: { filepath } }) {
-		// TODO
-		console.log(org, filepath);
+	async create({ org, name, params : { filepath } } = { params: { } }) {
+		const orgName = getOrgName(org);
+		const api = createAPI();
+		// get name from filepath
+		if (!filepath) {
+			// use default directory
+			filepath = '.';
+		}
+		if (!name) {
+			const question = {
+				type: 'input',
+				name: 'name',
+				message: 'What would you like to call your Function?'
+			};
+			const result =  await this.ui.prompt([question]);
+			name = result.name;
+		}
+		// trim name
+		name = name.trim();
+		// ask for description
+		const question = {
+			type: 'input',
+			name: 'description',
+			message: 'Add a description for your Function (optional)'
+		};
+		const result =  await this.ui.prompt([question]);
+		const description = result.description;
+		const slugName = name.toLowerCase().replace(/\s/g, '-');
+		const destinationPath = path.join(filepath, slugName);
+
+		this.ui.stdout.write(`Creating Logic Function ${this.ui.chalk.bold(name)} for ${orgName}...${os.EOL}`);
+		await this._validateExistingName({ api, org, name });
+		await this._validateExistingFiles({ templatePath: logicFunctionTemplatePath, destinationPath });
+		const createdFiles = await this._copyAndReplaceLogicFunction({
+			logicFunctionName: name,
+			logicFunctionSlugName: slugName,
+			description,
+			templatePath: logicFunctionTemplatePath,
+			destinationPath: path.join(filepath, slugName)
+		});
+		this.ui.stdout.write(`Successfully created ${this.ui.chalk.bold(name)} in ${this.ui.chalk.bold(filepath)}${os.EOL}`);
+		this.ui.stdout.write(`Files created:${os.EOL}`);
+		createdFiles.forEach((file) => {
+			this.ui.stdout.write(`- ${file}${os.EOL}`);
+		});
+		this.ui.stdout.write(`${os.EOL}Guidelines for creating your Logic Function can be found <TBD>.${os.EOL}`);
+		this.ui.stdout.write(`Once you have written your Logic Function, run${os.EOL}`);
+		this.ui.stdout.write(`- \`particle logic execute\` to run your Function${os.EOL}`);
+		this.ui.stdout.write(`- \`particle logic deploy\` to deploy your new changes${os.EOL}`);
+		return createdFiles;
+	}
+
+	async _validateExistingName({ api, org, name }) {
+		// TODO (hmontero): request for a getLogicFunctionByName() method in the API
+		let existingLogicFunction;
+		try {
+			const logicFunctionsResponse = await api.getLogicFunctionList({ org });
```

Comment:
IMO, This can directly use the `list` method from this class.

Until we may have an API to get the name and id separately, I added [this internal function](https://github.com/particle-iot/particle-cli/compare/feature/sc-123396/implement-execute-command-with-eventdata...feature/logic-function-get#diff-14f13d7b42c6c8b6494d7858d3a4191c54c8a528a3689709dd0ac6c2abbf92a7R96-R115) to fetch the full list of the logic functions, and obtain the `name` if `id` is given and vice versa. @hugomontero raised a concern about downloading a potentially _large_ list of logic functions and how that could impact the behavior or performance.

I made the above changes in the future branch: https://github.com/particle-iot/particle-cli/tree/feature/logic-function-get